### DISCUSSION TITLE
Custom Query erroring in Budibase SELECT @...

### DISCUSSION THREAD

**Discussion by mahxds** [Budibase/budibase#7775] [Help] [answered]
> set @prodid = 0, @lrid = 0, @lgortid =0;
> 
> SELECT @prodid := prod_id FROM tb_produkte WHERE prod_name = 'Name1';
> SELECT @lrid := lr_id FROM tb_lieferanten WHERE lr_name = 'Shipper1';
> SELECT @lgortid := lgort_id FROM tb_lagerorte WHERE lgort_name = 'Garage1';
> 
> INSERT INTO `tb_lager` (`lg_produkt_id`, `lg_lieferant_id`, `lg_lagerort_id`, `lg_menge`, `lg_palette`, `lg_ekpreis`)
> VALUES (@prodid , @lrid , @lgortid  , '77' , '77', '77');

Hi, im getting following error when i try this as an custom query .

> **Query Error: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'SELECT @prodid := prod_id FROM tb_produkte WHERE prod_name = 'Name1', SELE...' at line 3**

in SQL this query just works fine. i somehow dont get where the error is .

Anybody an Idea?

**Comment by melohagan** [comment, marked as answer]
Hey @mahxds 

I see what you mean. We're using the **mysql2** library, so I've raised a discussion there which you can follow: https://github.com/sidorares/node-mysql2/discussions/1636

**Reply by mahxds** [reply]
thank you for your help

**Reply by melohagan** [reply]
I've raised a bug here: https://github.com/Budibase/budibase/issues/7912
I'll look at it now.
